auth/main_test.go

```diff
@@ -0,0 +1,18 @@
+package auth
+
+import (
+	"os"
+	"testing"
+
+	"github.com/couchbase/sync_gateway/base"
+)
+
+func TestMain(m *testing.M) {
+	base.TestBucketPool = base.NewTestBucketPool(base.FlushBucketEmptierFunc, base.NoopInitFunc)
+
+	status := m.Run()
+
+	base.TestBucketPool.Close()
```

Comment:
In short no, but I'm going to change it anyway, as it does make the code a bit easier to read with defers in place.

The long version: The only case I can think of is when a test panics, but because of the way Go's test framework handles panics, they're recovered, and logged with os.Exit, which means deferred statements still aren't run.